### DISCUSSION TITLE
Add distributed rate limiting support via Redis

### DISCUSSION THREAD

**Discussion by dwhitlow** [Shopify/limiter#35] [Ideas]
Thanks for creating this project! The interface is intuitive, but I had a use case for it in mind that is better served with a Redis backend so RateQueue can share state across multiple processes or services.

I've hacked together an implementation based on the `redis` gem. If there's interest, I can refactor it so RateQueue supports various backends for state management and open a PR.

**Comment by dwhitlow** [comment]
Digging into this a bit more, I'll likely need to do some refactoring on Clock as well since the monotomic clock usage will present issues with a distributed approach. I'm still amenable to this if there's interest, though.

**Comment by sbfaulkner** [comment]
I think there are other packages out there that support using redis as a backend. This gem is intentionally simpler than those, so I dont' foresee us merging something like that into it.

Please feel free to fork a new gem inspired by / based on this one though.

Cheers
